Build error with FW_FILEID_ASSERT option

| | |
|:---|:---|
|**_F´ Version_**| |
|**_Affected Component_**|   |
---
## Problem Description

When enabling `FW_FILEID_ASSERT` and building a project, compilation fails because of the offending line

https://github.com/nasa/fprime/blob/51da1caf9640680779a4c595be00ca55717b9d28/Fw/Types/Assert.hpp#L22

The `#define FILE_NAME_ARG (U32)` is being injected as is in a function declaration, which reads `(U32)` as a type cast.

<details><summary>Error log</summary>

```
FAILED: F-Prime/default/config/CMakeFiles/default_config.dir/PolyDbEntryEnumAc.cpp.o
/Library/Developer/CommandLineTools/usr/bin/c++  -I/Users/chammard/Work/fp/fprime -I/Users/chammard/Work/fp/fprime/Ref/build-fprime-automatic-native -I/Users/chammard/Work/fp/fprime/Ref/build-fprime-automatic-native/F-Prime -I/Users/chammard/Work/fp/fprime/Ref/build-fprime-automatic-native/F-Prime/default/config/.. -I/Users/chammard/Work/fp/fprime/Ref/build-fprime-automatic-native/cmake/platform/unix/Platform/.. -std=c++14 -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX26.2.sdk -Wold-style-cast -Wall -Wconversion -Wdouble-promotion -Werror -Wextra -Wno-unused-parameter -Wno-vla -Wshadow -pedantic -DASSERT_FILE_ID=0x964c8301 -DASSERT_RELATIVE_PATH='"default/config/PolyDbEntryEnumAc.cpp"' -MD -MT F-Prime/default/config/CMakeFiles/default_config.dir/PolyDbEntryEnumAc.cpp.o -MF F-Prime/default/config/CMakeFiles/default_config.dir/PolyDbEntryEnumAc.cpp.o.d -o F-Prime/default/config/CMakeFiles/default_config.dir/PolyDbEntryEnumAc.cpp.o -c /Users/chammard/Work/fp/fprime/Ref/build-fprime-automatic-native/F-Prime/default/config/PolyDbEntryEnumAc.cpp
In file included from /Users/chammard/Work/fp/fprime/Ref/build-fprime-automatic-native/F-Prime/default/config/PolyDbEntryEnumAc.cpp:10:
/Users/chammard/Work/fp/fprime/Fw/Types/Assert.hpp:82:27: error: use of undeclared identifier 'file'; did you mean 'false'?
   82 | I8 SwAssert(FILE_NAME_ARG file, FwSizeType lineNo) NOINLINE CLANG_ANALYZER_NORETURN;
      |                           ^~~~
      |                           false
/Users/chammard/Work/fp/fprime/Fw/Types/Assert.hpp:82:13: error: use of old-style cast [-Werror,-Wold-style-cast]
   82 | I8 SwAssert(FILE_NAME_ARG file, FwSizeType lineNo) NOINLINE CLANG_ANALYZER_NORETURN;
      |             ^             ~~~~
/Users/chammard/Work/fp/fprime/Fw/Types/Assert.hpp:22:23: note: expanded from macro 'FILE_NAME_ARG'
   22 | #define FILE_NAME_ARG (U32)
      |                       ^
/Users/chammard/Work/fp/fprime/Fw/Types/Assert.hpp:82:33: error: unexpected type name 'FwSizeType': expected expression
   82 | I8 SwAssert(FILE_NAME_ARG file, FwSizeType lineNo) NOINLINE CLANG_ANALYZER_NORETURN;
      |                                 ^
/Users/chammard/Work/fp/fprime/Fw/Types/Assert.hpp:82:44: error: expected ')'
   82 | I8 SwAssert(FILE_NAME_ARG file, FwSizeType lineNo) NOINLINE CLANG_ANALYZER_NORETURN;
      |                                            ^
/Users/chammard/Work/fp/fprime/Fw/Types/Assert.hpp:82:12: note: to match this '('
   82 | I8 SwAssert(FILE_NAME_ARG file, FwSizeType lineNo) NOINLINE CLANG_ANALYZER_NORETURN;
      |            ^
/Users/chammard/Work/fp/fprime/Fw/Types/Assert.hpp:82:51: error: expected ';' after top level declarator
   82 | I8 SwAssert(FILE_NAME_ARG file, FwSizeType lineNo) NOINLINE CLANG_ANALYZER_NORETURN;
      |                                                   ^
      |                                                   ;
/Users/chammard/Work/fp/fprime/Fw/Types/Assert.hpp:82:52: error: declaration does not declare anything [-Werror,-Wmissing-declarations]
   82 | I8 SwAssert(FILE_NAME_ARG file, FwSizeType lineNo) NOINLINE CLANG_ANALYZER_NORETURN;
      |                                                    ^
/Users/chammard/Work/fp/fprime/Fw/Types/Assert.hpp:74:18: note: expanded from macro 'NOINLINE'
   74 | #define NOINLINE __attribute__((noinline))
      |                  ^
/Users/chammard/Work/fp/fprime/Fw/Types/Assert.hpp:85:4: error: redefinition of 'SwAssert'
   85 | I8 SwAssert(FILE_NAME_ARG file, FwAssertArgType arg1, FwSizeType lineNo) NOINLINE CLANG_ANALYZER_NORETURN;
      |    ^
/Users/chammard/Work/fp/fprime/Fw/Types/Assert.hpp:82:4: note: previous definition is here
   82 | I8 SwAssert(FILE_NAME_ARG file, FwSizeType lineNo) NOINLINE CLANG_ANALYZER_NORETURN;
      |    ^
/Users/chammard/Work/fp/fprime/Fw/Types/Assert.hpp:85:27: error: use of undeclared identifier 'file'; did you mean 'false'?
   85 | I8 SwAssert(FILE_NAME_ARG file, FwAssertArgType arg1, FwSizeType lineNo) NOINLINE CLANG_ANALYZER_NORETURN;
      |                           ^~~~
      |                           false
/Users/chammard/Work/fp/fprime/Fw/Types/Assert.hpp:85:13: error: use of old-style cast [-Werror,-Wold-style-cast]
   85 | I8 SwAssert(FILE_NAME_ARG file, FwAssertArgType arg1, FwSizeType lineNo) NOINLINE CLANG_ANALYZER_NORETURN;
      |             ^             ~~~~
/Users/chammard/Work/fp/fprime/Fw/Types/Assert.hpp:22:23: note: expanded from macro 'FILE_NAME_ARG'
   22 | #define FILE_NAME_ARG (U32)
      |                       ^
/Users/chammard/Work/fp/fprime/Fw/Types/Assert.hpp:85:33: error: unexpected type name 'FwAssertArgType': expected expression
   85 | I8 SwAssert(FILE_NAME_ARG file, FwAssertArgType arg1, FwSizeType lineNo) NOINLINE CLANG_ANALYZER_NORETURN;
      |                                 ^
/Users/chammard/Work/fp/fprime/Fw/Types/Assert.hpp:85:49: error: expected ')'
   85 | I8 SwAssert(FILE_NAME_ARG file, FwAssertArgType arg1, FwSizeType lineNo) NOINLINE CLANG_ANALYZER_NORETURN;
      |                                                 ^
/Users/chammard/Work/fp/fprime/Fw/Types/Assert.hpp:85:12: note: to match this '('
   85 | I8 SwAssert(FILE_NAME_ARG file, FwAssertArgType arg1, FwSizeType lineNo) NOINLINE CLANG_ANALYZER_NORETURN;
      |            ^
/Users/chammard/Work/fp/fprime/Fw/Types/Assert.hpp:85:73: error: expected ';' after top level declarator
   85 | I8 SwAssert(FILE_NAME_ARG file, FwAssertArgType arg1, FwSizeType lineNo) NOINLINE CLANG_ANALYZER_NORETURN;
      |                                                                         ^
      |                                                                         ;
/Users/chammard/Work/fp/fprime/Fw/Types/Assert.hpp:85:74: error: declaration does not declare anything [-Werror,-Wmissing-declarations]
   85 | I8 SwAssert(FILE_NAME_ARG file, FwAssertArgType arg1, FwSizeType lineNo) NOINLINE CLANG_ANALYZER_NORETURN;
      |                                                                          ^
/Users/chammard/Work/fp/fprime/Fw/Types/Assert.hpp:74:18: note: expanded from macro 'NOINLINE'
   74 | #define NOINLINE __attribute__((noinline))
      |                  ^
/Users/chammard/Work/fp/fprime/Fw/Types/Assert.hpp:88:4: error: redefinition of 'SwAssert'
   88 | I8 SwAssert(FILE_NAME_ARG file,
      |    ^
/Users/chammard/Work/fp/fprime/Fw/Types/Assert.hpp:82:4: note: previous definition is here
   82 | I8 SwAssert(FILE_NAME_ARG file, FwSizeType lineNo) NOINLINE CLANG_ANALYZER_NORETURN;
      |    ^
/Users/chammard/Work/fp/fprime/Fw/Types/Assert.hpp:88:27: error: use of undeclared identifier 'file'; did you mean 'false'?
   88 | I8 SwAssert(FILE_NAME_ARG file,
      |                           ^~~~
      |                           false
/Users/chammard/Work/fp/fprime/Fw/Types/Assert.hpp:88:13: error: use of old-style cast [-Werror,-Wold-style-cast]
   88 | I8 SwAssert(FILE_NAME_ARG file,
      |             ^             ~~~~
/Users/chammard/Work/fp/fprime/Fw/Types/Assert.hpp:22:23: note: expanded from macro 'FILE_NAME_ARG'
   22 | #define FILE_NAME_ARG (U32)
      |                       ^
/Users/chammard/Work/fp/fprime/Fw/Types/Assert.hpp:89:13: error: unexpected type name 'FwAssertArgType': expected expression
   89 |             FwAssertArgType arg1,
      |             ^
/Users/chammard/Work/fp/fprime/Fw/Types/Assert.hpp:89:29: error: expected ')'
   89 |             FwAssertArgType arg1,
      |                             ^
/Users/chammard/Work/fp/fprime/Fw/Types/Assert.hpp:88:12: note: to match this '('
   88 | I8 SwAssert(FILE_NAME_ARG file,
      |            ^
/Users/chammard/Work/fp/fprime/Fw/Types/Assert.hpp:91:31: error: expected ';' after top level declarator
   91 |             FwSizeType lineNo) NOINLINE CLANG_ANALYZER_NORETURN;
      |                               ^
      |                               ;
fatal error: too many errors emitted, stopping now [-ferror-limit=]
```
</details>

Introduced by https://github.com/nasa/fprime/pull/4568

## Context / Environment

Execute `fprime-util version-check` and share the output.

```
fprime-util version-check                                                                                                                                                                                                                                                                                                                               
Operating System: Darwin
CPU Architecture: arm64
Platform: macOS-26.3-arm64-arm-64bit
Python version: 3.12.8
CMake version: 3.26.0
Pip version: 26.0.1
Pip packages:
    fprime-tools==4.1.0
    fprime-gds==4.1.1a5.dev10+g0ffcaf601.d20260223
    fprime-fpp==3.1.1a4
Project submodules:
```

## How to Reproduce

1. checkout `fprime@devel`
2. Set `#define FW_ASSERT_LEVEL (FW_FILEID_ASSERT)`
3. build fprime/Ref

--- a/Fw/Types/Assert.hpp
+++ b/Fw/Types/Assert.hpp
@@ -19,13 +19,13 @@
 // Passing the __LINE__ argument at the end of the function ensures that
 // the FW_ASSERT_NO_FIRST_ARG macro will never have an empty variadic variable
 #if FW_ASSERT_LEVEL == FW_FILEID_ASSERT && defined ASSERT_FILE_ID
-#define FILE_NAME_ARG (U32)
+#define FILE_NAME_ARG U32
 #define FW_ASSERT(...)                            \
     ((void)((FW_ASSERT_FIRST_ARG(__VA_ARGS__, 0)) \
                 ? (0)                             \
                 : (Fw::SwAssert(ASSERT_FILE_ID, FW_ASSERT_NO_FIRST_ARG(__VA_ARGS__, __LINE__)))))
 #elif FW_ASSERT_LEVEL == FW_FILEID_ASSERT && !defined ASSERT_FILE_ID
-#define FILE_NAME_ARG (U32)
+#define FILE_NAME_ARG U32
 #define FW_ASSERT(...)                            \
     ((void)((FW_ASSERT_FIRST_ARG(__VA_ARGS__, 0)) \
                 ? (0)                             \

